src/widgets/BaseWindow.cpp

```diff
@@ -715,6 +742,86 @@ void BaseWindow::updateScale()
     }
 }
 
+void BaseWindow::handleMinimizeEvent(QEvent *event)
+{
+    auto showMessage = [&] {
+        QMessageBox msgBox(QMessageBox::Question, "Chatterino",
+                           "What should we do when you press the "
+                           "minimize button? Minimizing to the taskbar "
+                           "is the default behavior.",
+                           QMessageBox::NoButton, this);
+        auto *taskbar =
+            msgBox.addButton("Minimize to task bar", QMessageBox::YesRole);
+        auto *tray =
+            msgBox.addButton("Minimize to tray icon", QMessageBox::ApplyRole);
+
+        msgBox.setWindowModality(Qt::ApplicationModal);
+        msgBox.setWindowFlags(msgBox.windowFlags().setFlag(
+            Qt::WindowContextHelpButtonHint, false));
+        msgBox.exec();
+
+        auto *result = msgBox.clickedButton();
```

Comment:
Initially the message box used the YesRole and NoRole. I assume that when as message box as an abort role it can be closed. Now that it only has YesRole and ApplyRole it can't be closed anymore